text/0000-cargo-sbom.md

```diff
@@ -0,0 +1,116 @@
+
+The SBOM file will be written to disk before `rustc` is executed for the each artifact. This enables [`RUSTC_WORKSPACE_WRAPPER`](https://doc.rust-lang.org/cargo/reference/config.html#buildrustc-workspace-wrapper) to point at a program that can utilize the SBOM file to embed information into the binary if desired.
+
+## Format
+The format will use JSON, but the exact format is not specified in this RFC. Additional fields can be added as needed.
+
+
+### Resolved Dependency Tree
+The SBOM will include the following information (if available) for each crate:
+- ID (opaque identifier)
+- Name
+- Version
+- Source (registry / git / path etc.)
+- Checksum 
+- Dependencies (list of IDs)
+- Type (normal, build)
+- Activated features
+
+If a crate is used as both a normal dependency and a build dependency that is separately compiled, then separate entries will exist in the dependency tree with the correct activated features listed for each instance.
+
+Checksum is an optional field, since only crates from registries have checksums. If a checksum is needed for a crate that comes from a path dependency for example, it will be up to the post-processing tool to produce an appropriate value.
+
+If further information is needed (such as license), then the post-processing tool can use `cargo metadata` or another mechanism to find it.
+
+### Resolved build confugration
+- Rust toolchain version
+- `RUSTFLAGS`
+- Current build profile name
+- Selected profile values
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+It introduces yet another SBOM format. However, the format is specifically designed to be used as an intermediate, to be converted to an industry-standard format by external tooling.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+Since there is no consensus on a single SBOM format within the software industry, and existing formats are still evolving, Cargo should not pick an existing SBOM format. If Cargo were to use existing SBOM formats, multiple formats (and multiple versions of each format) would need to be supported. The task of generating a specific SBOM format is best left to applications outside Cargo or Cargo extension.
+
+Unfortunately it's difficult to extract accurate SBOM information with existing options. Using the `Cargo.lock` file or `cargo metadata` overincludes dependencies. Additionally, since Cargo has many different commands that produce compiled artifacts (build, test, bench, etc.) and each of these commands take arguments that can affect the dependency list it's difficult to ensure that the correct dependency list is used.
+
+Adding an option to `cargo metadata` to support resolver v2 would help with overinclusion of dependencies, but still makes it difficult to ensure the exact set of features, command-line arguments, and other options are taken into account. Additionally, since build scripts (`build.rs`) may impact the output, `cargo metadata` may need to execute them.
```

Comment:
imo this undersells the problem with `cargo metadata`.

To properly support resolver v2, `cargo metadata` would need to have a CLI more akin to `cargo check` which includes
- Package selection
- Build target selection within a package
- etc

which some of this would require either breaking changes to `cargo metadata` or making the CLI inconsistent from every other cargo subcommand.